cmd/cluster/create.go

```diff
@@ -157,7 +165,18 @@ func CreateCluster(ctx context.Context, opts Options) error {
 	client := util.GetClientOrDie()
 
 	// Load or create infrastructure for the cluster
+	switch opts.Platform {
+	case "aws", string(v1alpha1.AWSPlatform):
+		return createInfraAWS(ctx, opts, pullSecret, sshKey, client, annotations)
+	case "none", "baremetal", string(v1alpha1.NonePlatform):
+		return createInfraNone(ctx, opts, pullSecret, sshKey, client, annotations)
+	}
+	return nil
+}
+
+func createInfraAWS(ctx context.Context, opts Options, pullSecret []byte, sshKey []byte, client crclient.Client, annotations map[string]string) error {
 	var infra *awsinfra.CreateInfraOutput
+	var err error
```

Comment:
why is this needed?

Reply:
error initialization for "infra, err = opt.CreateInfra(ctx)"

Reply:
i also don't see why this is needed